src/stackdriver_agent.c

```diff
@@ -206,9 +214,25 @@ static int sagt_init()
     return result;
 }
 
+static int sagt_config(const char *key, const char *value) {
+  if (strcmp(key, "Hostname") == 0) {
+    hostname = (const char *) sstrdup(value);
+    if (hostname == NULL) {
+      WARNING("%s: sagt_config sstrdup failed. Using defaults.",
```

Comment:
I think you actually want to fail here, rather than silently fall back on the global hostname.